lib/src/main/java/com/basistheory/android/util/DataManipulationUtils.kt

```diff
@@ -41,8 +42,8 @@ internal fun replaceElementRefs(value: Any?): Any? {
         }
     } else {
         when (value) {
-            is TextElement -> value.tryGetTextToTokenize()
-            is ElementValueReference -> value.getValue()
+            is TextElement -> value.tryGetTextToTokenize().toValueType(value.getValueType)
+            is ElementValueReference -> value.getValue().toValueType(value.getValueType)
```

Comment:
Couldn't we make this a little cleaner by always converting to the value type inside `tryGetTextToTokenize()` and `getValue()`? Then we wouldn't need to chain `.toValueType` on all these calls, passing in another property off of the element when that is already encapsulated inside of `value`.